test/ExecutionTests.jl

```diff
@@ -148,12 +148,38 @@ tune!(b_pass)
 # warmup #
 ###########
 
-p = params(warmup(@benchmarkable sin(1)))
+@test_deprecated warmup(@benchmarkable sin(1))
+
+is_warm = false
+function needs_warm()
+    global is_warm
+    if is_warm
+        sleep(0.1)
+    else
+        sleep(2)
+        is_warm = true
+    end
+end
+
+w = @benchmarkable needs_warm()
+w.params.seconds = 1
+
+#test that all measurements from lineartrial used in tune! are warm
+is_warm = false
+@test maximum(BenchmarkTools.lineartrial(w, w.params)) < 1e9
+
+#test that run warms up the benchmark
+tune!(w)
+is_warm = false
+@test minimum(run(w).times) < 1e9
```

Comment:
why not `maximum`?

Reply:
This goes back to the reason I filed the original issue. If you have a benchmark that takes longer than 5 seconds to compile, then not warming up and calling run on the benchmark leaves you with a measurements vector whose only entry is measuring compile time. Since my code was reducing the benchmarks with `minimum` this effect was visible. However, I agree that this would be much clearer if we set it to maximum.

Reply:
When we test the output of `@belapsed`, we are using exactly the same test as here, since that also calls minimum. We're just trying to figure out if only the cold execution was measured, the failure case I said above.